src/backend/distributed/planner/multi_router_planner.c

```diff
@@ -681,6 +684,54 @@ MostPermissiveVolatileFlag(char left, char right)
 
 
 /*
+ * TargetEntryChangesValue determines whether the given target entry may
+ * change the value in a given column, given a join tree. The result is
+ * true if the expression refers directly to the column or is implied by
+ * the qualifiers of the join tree, and false otherwise.
```

Comment:
Is this comment wrong?

I think it should be: "The result is true if the expression refers directly to the column and is not implied by the qualifiers of the join tree, and false otherwise."